tests/test_using.py

```diff
@@ -555,6 +677,50 @@ class Meta:
         test_model = TestModel2Factory()
         self.assertEqual(4, test_model.two.three)
 
+    def test_self_attribute_parent_async(self):
+        class Book(FakeAsyncModel):
+            pass
+
+        class Author(FakeAsyncModel):
+            pass
+
+        class Chapter(FakeAsyncModel):
+            pass
+
+        class AuthorFactory(FakeAsyncModelFactory):
+            class Meta:
+                model = Author
+
+            hometown = "Paris"
+
+        class ChapterFactory(FakeAsyncModelFactory):
+            class Meta:
+                model = Chapter
+
+            author = factory.SubFactory(AuthorFactory)
+
+        class BookFactory(FakeAsyncModelFactory):
+            class Meta:
+                model = Book
+
+            author = factory.SubFactory(AuthorFactory, hometown="Toronto")
+            chapter = factory.SubFactory(ChapterFactory, author=factory.SelfAttribute('..author'))
+            preface = factory.SubFactory(ChapterFactory)
+
+        loop = asyncio.get_event_loop()
+        book = loop.run_until_complete(BookFactory())
+
+        self.assertEqual(book.author, book.chapter.author)
+        self.assertEqual("Toronto", book.author.hometown)
+        self.assertEqual("Paris", book.preface.author.hometown)
+        self.assertEqual(2, book.id)
+        self.assertEqual(2, book.author.id)
+        self.assertEqual("Toronto", book.author.hometown)
+        self.assertEqual(2, book.chapter.id)
+        self.assertEqual(2, book.preface.id)
+        self.assertEqual(2, book.preface.author.id)
```

Comment:
This is not meaningful. The preface author would actually be a different object in the DB and have a different id from the book author. One is from Paris, the other from Toronto.

IIUC, the ids are checked to make sure the `create()` method of `FakeAsyncModel` has been called? If so, thoughts on adding a comment and/or clarifying the magic value with a unique object for `id`? For example:
```python
create_marker = object()

class FakeAsyncModel:
    @classmethod
    async def create(cls, **kwargs):
        instance = cls(**kwargs)
        if not instance.id:
            instance.id = create_marker
        # ...

        # In the tests:
        self.assertIs(book.author.id, create_marker)
```
?

Reply:
> IIUC, the ids are checked to make sure the create() method of FakeAsyncModel has been called?
exact.
I'll improve that